### ISSUE TITLE
CLIP Score source code

### ISSUE THREAD

**Issue by EadCat** [GongyeLiu/StyleCrafter#5]
Can you provide the CLIP score evaluation code?

Thank you.

**Comment by GongyeLiu** [comment]
Hi, sorry for my late late reply(missing for check).  We're using [CLIP ViT-H/14 - LAION-2B](https://huggingface.co/laion/CLIP-ViT-H-14-laion2B-s32B-b79K) to extract image/text embeddings and calculate the cosine similarity(and further calculate the average). Here's an example code:

```python
import open_clip
import torch
from PIL import Image

device = 'cuda'
# load model
model, _, process = open_clip.create_model_and_transforms('ViT-H-14',pretrained='laion2b_s32b_b79k', device=device)
tokenizer = open_clip.get_tokenizer('ViT-H-14')

# extract text emb
text = tokenizer("Example Caption.").to(device)
with torch.no_grad():
    text_features = model.encode_text(text).float()
text_features /= text_features.norm(dim=-1, keepdim=True)

# extract image emb
image = Image.open('example.jpg')
image_input = process(image).unsqueeze(0).to(device)
with torch.no_grad():
    image_features = model.encode_image(image_input).float()
image_features /= image_features.norm(dim=-1, keepdim=True)

# calculate similarity
similarity = torch.cosine_similarity(text_features, image_features, dim=-1).mean().item()
print(similarity)
```

or refer to the official example in open_clip's [Colab](https://colab.research.google.com/github/mlfoundations/open_clip/blob/master/docs/Interacting_with_open_clip.ipynb).

Hope this helps u

**Comment by EadCat** [comment]
I understand that this can be utilized for measuring CLIP scores between text and images. How was the metric 'Style Conformity' measured in the paper? Would you give me a clue?

**Comment by GongyeLiu** [comment]
We measure "Style Conformity" by calculating the CLIP similarity between generated results and input style reference, as most style-guided methods do.

**Comment by EadCat** [comment]
So, do you mean that you measured the similarity by encoding both the reference image and the generated results to feature maps?

**Comment by GongyeLiu** [comment]
yeah, that's correct.

**Comment by EadCat** [comment]
I got it. Thank you for the great work!
